server-side/index.js

```diff
@@ -112,13 +112,29 @@ app.post('/api/keys', async (request, response) => {
     }
 });
 
-app.delete('/api/keys', async (request, response) => {
+app.update('/api/keys', async (request, response) => {
+    const action = request.query.action;
+
     try {
         const userId = await jwt.getUserId(request.cookies.jwt);
-        await database.query('update `keys` set `revoked_at` = ? where `key` = ?', [
-            new Date(),
-            request.query.key || '',
-        ]);
+
+        switch (action) {
+            case 'revoke':
+                await database.query('update `keys` set `revoked_at` = ? where `key` = ?', [
+                    new Date(),
+                    request.query.key || '',
+                ]);
+                break;
+            case 'reinstate':
+                await database.query('update `keys` set `revoked_at` = ? where `key` = ?', [
+                    null,
+                    request.query.key || '',
+                ]);
+                break;
+            default:
+                // unknown command
+                break;
+        }
```

Comment:
With the RESTful approach we can just save the model which will be a bit cleaner than the switch.